src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs

```diff
@@ -349,13 +361,19 @@ internal void VerifyIncomingMessage(ref Message message, TimeSpan timeout)
         public abstract Task DispatchAsync(Message message);
     }
 
-    internal class SecurityReplyChannelDispatcher : ServerSecurityChannelDispatcher<IReplyChannel>  //, IChannel
+    internal class SecurityReplyChannelDispatcher : ServerSecurityChannelDispatcher<IReplyChannel>, IReplyChannel
     {
         private readonly bool _sendUnsecuredFaults;
         internal static readonly SecurityStandardsManager defaultStandardsManager = SecurityStandardsManager.DefaultInstance;
 
-        public SecurityReplyChannelDispatcher(SecurityServiceDispatcher securityServiceDispatcher, IReplyChannel innerChannel, SecurityProtocol securityProtocol)
-                : base(securityServiceDispatcher, innerChannel, securityProtocol)
+        public event EventHandler Closed;
+        public event EventHandler Closing;
```

Comment:
There's going to be a compiler warning. Please add a pragma and open an issue that this class needs to derive from CommunicationObject of some form, probably SerivceChannelBase